File: backend_core_mobile/app/models/cliente_model.py
# cliente_model.py
import uuid
from sqlalchemy import Column, String, Integer, DateTime, ForeignKey, Date, Numeric
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship
from app.database.session import Base

class Cliente(Base):
    __tablename__ = "clientes"

    id_cliente = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    id_usuario = Column(UUID(as_uuid=True), ForeignKey("usuarios.id_usuario"), nullable=True)
    id_agencia = Column(UUID(as_uuid=True), ForeignKey("agencias.id_agencia"), nullable=True)
    documento = Column(String(15), unique=True, nullable=False)
    nombres = Column(String(100), nullable=False)
    apellidos = Column(String(100), nullable=False)
    telefono = Column(String(20))
    correo = Column(String(120))
    direccion = Column(String)
    distrito = Column(String(100))
    provincia = Column(String(100))
    departamento = Column(String(100))
    fecha_nacimiento = Column(Date)
    estado_civil = Column(String(30))
    ocupacion = Column(String(100))
    tipo_cliente = Column(String(30))
    estado = Column(String(20), default="ACTIVO")
    created_at = Column(DateTime(timezone=True))
    updated_at = Column(DateTime(timezone=True))

    usuario = relationship("Usuario", foreign_keys=[id_usuario])
    agencia = relationship("Agencia", foreign_keys=[id_agencia])
    negocios = relationship("NegocioCliente", back_populates="cliente", cascade="all, delete-orphan")

class NegocioCliente(Base):
    __tablename__ = "negocios_cliente"

    id_negocio = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    id_cliente = Column(UUID(as_uuid=True), ForeignKey("clientes.id_cliente"), nullable=False)
    nombre_comercial = Column(String(150), nullable=False)
    giro_negocio = Column(String(100))
    antiguedad_meses = Column(Integer)
    ingreso_mensual = Column(Numeric(12, 2))
    gasto_mensual = Column(Numeric(12, 2))
    direccion_negocio = Column(String)
    lat_negocio = Column(Numeric(10, 7))
    lng_negocio = Column(Numeric(10, 7))
    estado = Column(String(20), default="ACTIVO")
    created_at = Column(DateTime(timezone=True))
    updated_at = Column(DateTime(timezone=True))

    cliente = relationship("Cliente", back_populates="negocios")

File: mobile_app_bcp/lib/shared/models/cliente_model.dart
// shared/models/cliente_model.dart
class ClienteModel {
  final String idCliente;
  final String idUsuario;
  final String? idAgencia;
  final String documento;
  final String nombres;
  final String apellidos;
  final String? telefono;
  final String? correo;
  final String? direccion;
  final String? distrito;
  final String? provincia;
  final String? departamento;
  final DateTime? fechaNacimiento;
  final String? estadoCivil;
  final String? ocupacion;
  final String? tipoCliente;
  final String estado;

  ClienteModel({
    required this.idCliente,
    required this.idUsuario,
    this.idAgencia,
    required this.documento,
    required this.nombres,
    required this.apellidos,
    this.telefono,
    this.correo,
    this.direccion,
    this.distrito,
    this.provincia,
    this.departamento,
    this.fechaNacimiento,
    this.estadoCivil,
    this.ocupacion,
    this.tipoCliente,
    this.estado = 'ACTIVO',
  });

  String get nombreCompleto => '$nombres $apellidos';

  factory ClienteModel.fromJson(Map<String, dynamic> json) {
    return ClienteModel(
      idCliente: json['id_cliente'] ?? '',
      idUsuario: json['id_usuario'] ?? '',
      idAgencia: json['id_agencia'],
      documento: json['documento'] ?? '',
      nombres: json['nombres'] ?? '',
      apellidos: json['apellidos'] ?? '',
      telefono: json['telefono'],
      correo: json['correo'],
      direccion: json['direccion'],
      distrito: json['distrito'],
      provincia: json['provincia'],
      departamento: json['departamento'],
      fechaNacimiento: json['fecha_nacimiento'] != null
          ? DateTime.tryParse(json['fecha_nacimiento'])
          : null,
      estadoCivil: json['estado_civil'],
      ocupacion: json['ocupacion'],
      tipoCliente: json['tipo_cliente'],
      estado: json['estado'] ?? 'ACTIVO',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id_cliente': idCliente,
      'id_usuario': idUsuario,
      'id_agencia': idAgencia,
      'documento': documento,
      'nombres': nombres,
      'apellidos': apellidos,
      'telefono': telefono,
      'correo': correo,
      'direccion': direccion,
      'distrito': distrito,
      'provincia': provincia,
      'departamento': departamento,
      'fecha_nacimiento': fechaNacimiento?.toIso8601String(),
      'estado_civil': estadoCivil,
      'ocupacion': ocupacion,
      'tipo_cliente': tipoCliente,
      'estado': estado,
    };
  }
}
